text/3463-crates-io-policy-update.md

```diff
@@ -0,0 +1,233 @@
+- Start Date: 2023-07-24
+- RFC PR: [rust-lang/rfcs#3463](https://github.com/rust-lang/rfcs/pull/3463)
+
+# Summary
+[summary]: #summary
+
+The Rust community has outgrown the current crates.io policies. This RFC proposes a new "Terms of Use" policy based on prior work by PyPI, npm and GitHub.
+
+# Motivation
+[motivation]: #motivation
+
+> Why are we doing this? What use cases does it support? What is the expected outcome?
+
+crates.io has a "[Package Policies](https://crates.io/policies)" page, which describes the current, organically grown policies. A lot of support requests or questionable uses of crates.io however need explicit decisions from the crates.io team, since many cases are currently not covered by these policies. Additionally, decisions made by the team may be seen as arbitrary without written guidelines.
+
+The situation around name squatting has lately also reached unsustainable levels, and while [namespaces](https://github.com/rust-lang/rfcs/pull/3243) might help with some parts of the problem we would still need policies for name squatting namespaces then.
+
+The main motivation for this RFC is to give the crates.io team a fixed set of rules to determine if a project is using crates.io in a reasonable way, or if the user should get a warning and the project potentially be removed. It is mostly codifying the existing practices of the team, except for being more strict regarding name squatting.
+
+# Proposal
+[guide-level-explanation]: #guide-level-explanation
+
+The following is a proposed new "Terms of Use" policy for crates.io, replacing <https://crates.io/policies> and <https://crates.io/data-access>.
+
+-------------------------------------------------------------------------------
+
+# Terms of Use
+
+**Short version:** _crates.io is a critical resource for the Rust ecosystem,
+which hosts a variety of packages from a diverse group of users. That resource
+is only effective when our users are able to work together as part of a
+community in good faith. While using crates.io, you must comply with our
+Acceptable Use Policies, which include some restrictions on content and conduct
+on crates.io related to user safety, intellectual property, privacy,
+authenticity, and other limitations. In short, be excellent to each other!_
+
+We do not allow content or activity on crates.io that:
+
+- violates the [Code of Conduct](https://www.rust-lang.org/policies/code-of-conduct)
+  of the Rust project
+- is unlawful or promotes unlawful activities, incurring legal liability in the
+  countries the Rust Foundation officially operates in
+- is libelous, defamatory, or fraudulent
+- amounts to phishing or attempted phishing
+- infringes any proprietary right of any party, including patent, trademark,
```

Comment:
This seems incredibly broad. For example, would a Rust port of ffmpeg, which contains a number of patented codecs (c.f. https://en.wikipedia.org/wiki/FFmpeg#Legal_aspects), be allowed on crates.io? What about a Rust clone of youtube-dl, which could be described as having "anti-circumvention code" (c.f. https://freedom.press/news/riaa-github-youtube-dl-journalist-tool/)?

This isn't hypothetical, there are bindings for libdvdcss already on crates.io: https://crates.io/crates/dvdcss. VLC [describes](https://wiki.videolan.org/Frequently_Asked_Questions/#Is_libdvdcss_legal.3F) the legality of their library as: 
> The use and distribution of the libdvdcss library is controversial in a few countries such as the United States because of a law called the DMCA (Digital Millennium Copyright Act). If you are unsure about the legality of using and distributing this library in your country, please consult your lawyer.

It's probably likely that a number of https://crates.io/search?q=mp4 are infringing upon MPEG LA patents as well.

Reply:
That is up for lawyers to decide and if the crates.io team gets such a request it will be forwarded to the legal team of the Rust Foundation. This is already the case today though, so nothing here is really changing in that regard.